sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RequestData.cs

```diff
@@ -27,12 +27,16 @@ public RequestData(int version, Activity activity, ref TagEnumerationState monit
                     break;
             }
 
-            Id = activity.Context.SpanId.ToHexString();
-            Name = TraceHelper.GetOperationName(activity, ref monitorTags.MappedTags);
-            Duration = activity.Duration.ToString("c", CultureInfo.InvariantCulture);
+            Id = activity.Context.SpanId.ToHexString().Truncate(SchemaConstants.RequestData_Id_MaxLength);
+            Name = TraceHelper.GetOperationName(activity, ref monitorTags.MappedTags).Truncate(SchemaConstants.RequestData_Name_MaxLength);
+            Duration = activity.Duration < SchemaConstants.RequestData_Duration_LessThanDays
+                ? activity.Duration.ToString("c", CultureInfo.InvariantCulture)
+                : SchemaConstants.Duration_MaxValue;
             Success = activity.GetStatus().StatusCode != StatusCode.Error;
-            ResponseCode = AzMonList.GetTagValue(ref monitorTags.MappedTags, SemanticConventions.AttributeHttpStatusCode)?.ToString() ?? "0";
-            Url = url;
+            ResponseCode = AzMonList.GetTagValue(ref monitorTags.MappedTags, SemanticConventions.AttributeHttpStatusCode)
+                ?.ToString().Truncate(SchemaConstants.RequestData_ResponseCode_MaxLength)
+                ?? "0";
+            Url = url.Truncate(SchemaConstants.RequestData_Url_MaxLength);
```

Comment:
What would happen if this truncation results in a mal-formed URL?

Reply:
the max length is 2048.
I think it's rare that this would occur, but if it does we would ingest an incomplete url.

I don't see any special instructions in the Mapping doc to handle this differently.